File: _code-samples/remove-regular-key/py/remove-regular-key.py
import json
from xrpl.clients import JsonRpcClient
from xrpl.wallet import generate_faucet_wallet, Wallet
from xrpl.models.transactions import SetRegularKey
from xrpl.models.requests import AccountInfo
from xrpl.transaction import submit_and_wait

client = JsonRpcClient("https://s.altnet.rippletest.net:51234")

print("Funding new wallet from faucet...")
wallet = generate_faucet_wallet(client)
print(f"""Funded. Master key pair:
  Address: {wallet.address}
  Seed: {wallet.seed}
""")

# Generate a regular key and assign it to the account -------------------------
# Skip this step if you are using a pre-existing account that already has a
# regular key configured.
algorithm = "ed25519"
regular_key_pair = Wallet.create(algorithm)
print(f"""Generated regular key pair:
  Address: {regular_key_pair.address}
  Seed: {regular_key_pair.seed}
  Algorithm: {algorithm}
""")
regular_key_tx = SetRegularKey(
    account=wallet.address, regular_key=regular_key_pair.address
)

print("Assigning regular key to the account...")
try:
    response = submit_and_wait(regular_key_tx, client, wallet)
except err:
    print("Submitting SetRegularKey transaction failed with error", err)
    exit(1)
set_regular_key_result_code = response.result["meta"]["TransactionResult"]

if set_regular_key_result_code == "tesSUCCESS":
    print("Regular Key set successfully.")
else:
    print(f"SetRegularKey failed with code {set_regular_key_result_code}.")
    exit(1)

# Check regular key associated with account -----------------------------------
account_info_resp = client.request(
    AccountInfo(account=wallet.address, ledger_index="validated")
)
if not account_info_resp.is_successful():
    print(f"Error looking up account:", account_info_resp.result)
    exit(1)

account_data = account_info_resp.result["account_data"]
print(f"Account info for {wallet.address}:")
print(json.dumps(account_data, indent=2))

if "RegularKey" in account_data.keys():
    print("Current regular key:", account_data["RegularKey"])
else:
    print("❌ No regular key set.")
    exit(1)

# Remove regular key from account ---------------------------------------------
remove_regular_key_tx = SetRegularKey(
    account=wallet.address
    # Omit regular_key field to remove existing regular key from account
)

print("Removing regular key from account...")
# When removing, you can sign with the regular key or master key
remove_resp = submit_and_wait(remove_regular_key_tx, client, regular_key_pair)
remove_regular_key_result_code = remove_resp.result["meta"]["TransactionResult"]

if remove_regular_key_result_code == "tesSUCCESS":
    print("Regular Key successfully removed.")
else:
    print("SetRegularKey (removing) failed with code", remove_regular_key_result_code)
    exit(1)

# Confirm that the account has no regular key ---------------------------------
account_info_resp2 = client.request(
    AccountInfo(account=wallet.address, ledger_index="validated")
)
if not account_info_resp2.is_successful():
    print("Error looking up account:", account_info_resp2.result)
    exit(1)

account_data2 = account_info_resp2.result["account_data"]
print(f"Account info for {wallet.address}:")
print(json.dumps(account_data2, indent=2))

if "RegularKey" in account_data2.keys():
    print("❌ Regular key address is:", account_data2["RegularKey"])
else:
    print("✅ No regular key set.")

File: _code-samples/remove-regular-key/js/remove-regular-key.js
import xrpl from 'xrpl'

const client = new xrpl.Client('wss://s.altnet.rippletest.net:51233')
await client.connect()

console.log('Funding new wallet from faucet...')
const { wallet } = await client.fundWallet()
console.log(`Funded. Master key pair:
  Address: ${wallet.address}
  Seed: ${wallet.seed}
`)

// Generate a regular key and assign it to the account -------------------------
// Skip this step if you are using a pre-existing account that already has a
// regular key configured.
const algorithm = 'ed25519'
const regularKeyPair = xrpl.Wallet.generate(algorithm)
console.log(`Generated regular key pair:
  Address: ${regularKeyPair.address}
  Seed: ${regularKeyPair.seed}
  Algorithm: ${algorithm}
`)
const regularKeyTx = {
  TransactionType: 'SetRegularKey',
  Account: wallet.address,
  RegularKey: regularKeyPair.address
}
xrpl.validate(regularKeyTx)

console.log('Assigning regular key to the account...')
const response = await client.submitAndWait(regularKeyTx, { wallet, autofill: true })
const setRegularKeyResultCode = response.result.meta.TransactionResult
if (setRegularKeyResultCode === 'tesSUCCESS') {
  console.log('Regular Key set successfully.')
} else {
  console.error(`SetRegularKey failed with code ${setRegularKeyResultCode}.`)
  client.disconnect()
  process.exit(1)
}

// Check regular key associated with account -----------------------------------
const accountInfoResp = await client.request({
  command: 'account_info',
  account: wallet.address,
  ledger_index: 'validated'
})
if (accountInfoResp.error) {
  console.error('Error looking up account:', accountInfoResp.error)
  client.disconnect()
  process.exit(1)
}
console.log(`Account info for ${wallet.address}:`)
console.log(JSON.stringify(accountInfoResp.result.account_data, null, 2))
if (accountInfoResp.result.account_data.RegularKey) {
  console.log('Current regular key:',
    accountInfoResp.result.account_data.RegularKey
  )
} else {
  console.log('No regular key set.')
  client.disconnect()
  process.exit(1)
}

// Remove regular key from account ---------------------------------------------
const removeRegularKeyTx = {
  TransactionType: 'SetRegularKey',
  Account: wallet.address
  // Omit RegularKey field to remove existing regular key from account
}
xrpl.validate(removeRegularKeyTx)

console.log('Removing regular key from account...')
const removeResp = await client.submitAndWait(removeRegularKeyTx, {
  wallet: regularKeyPair, // When removing, you can use the regular key or master key
  autofill: true
})
const removeRegularKeyResultCode = removeResp.result.meta.TransactionResult
if (removeRegularKeyResultCode === 'tesSUCCESS') {
  console.log('Regular Key successfully removed.')
} else {
  console.error('SetRegularKey (removing) failed with code', 
    removeRegularKeyResultCode
  )
  client.disconnect()
  process.exit(1)
}

// Confirm that the account has no regular key ---------------------------------
const accountInfoResp2 = await client.request({
  command: 'account_info',
  account: wallet.address,
  ledger_index: 'validated'
})
if (accountInfoResp2.error) {
  console.error('Error looking up account:', accountInfoResp2.error)
  client.disconnect()
  process.exit(1)
}
console.log(`Account info for ${wallet.address}:`)
console.log(JSON.stringify(accountInfoResp2.result.account_data, null, 2))
if (accountInfoResp2.result.account_data.RegularKey) {
  console.log('Regular key address is:', 
    accountInfoResp2.result.account_data.RegularKey
  )
} else {
  console.log('No regular key set.')
}

client.disconnect()
